integration-tests/regression.test.js

```diff
@@ -232,11 +237,140 @@ describe('Tram-One', () => {
 
 	it('should trigger use-effects of the first resolved element', async () => {
 		// start the app
-		startApp();
+		await startAppAndWait();
 
 		// previously, useEffects on the first resolved element would not trigger
 		// because the effect queue and effect store were pointed to the same object instance
 
 		expect(document.title).toEqual('Tram-One Testing App');
 	});
+
+	it('should keep focus on inputs without a start and end selection', async () => {
+		// start the app
+		const { container } = await startAppAndWait();
+
+		// previously when interacting with an input of a different type (e.g. range)
+		// when reapplying focus Tram-One would throw an error because while the
+		// function for setting selection range exists, it does not work
+
+		// focus on the input (the range input defaults to 0)
+		userEvent.click(getByLabelText(container, 'Store Generator'));
+
+		// verify that the element has focus (before changing the value)
+		await waitFor(() => {
+			expect(getByLabelText(container, 'Store Generator')).toHaveFocus();
+		});
+
+		// change the value of the input
+		fireEvent.change(getByLabelText(container, 'Store Generator'), { target: { value: 1 } });
```

Comment:
I actually wish we could do a `userEvent` here, but sadly the arrow interactions don't update the input value. If we wanted to make this work we'd have to make a change to the user-events repo - something similar to this: https://github.com/testing-library/user-event/blob/ee062e762f9ac185d982dbf990387e97e05b3c9d/src/pointer/pointerPress.ts#L254-L327